odoo/addons/base/models/ir_actions.py

```diff
@@ -564,17 +564,17 @@ def _run_action_multi(self, eval_context=None):
             res = act.run() or res
         return res
 
-    def _run_action_object_write(self, eval_context=None):
-        """Apply specified write changes to active_id."""
+    def _run_action_object_write_multi(self, eval_context=None):
+        """Apply specified write changes to active_ids."""
         vals = self.fields_lines.eval_value(eval_context=eval_context)
         res = {line.col1.name: vals[line.id] for line in self.fields_lines}
 
-        if self._context.get('onchange_self'):
+        if len(self) == 1 and self._context.get('onchange_self'):
```

Comment:
IMO this condition does not make sense, as `self` is not the recordset on which the action is applied, it is the action itself.  And by design `self` is always a singleton.